diff --git a/dist/hls.js b/dist/hls.js
index 5582eb1ed0457ad6e894be234ad928d79152d933..cd6238e5ac6373c1fe381daf48f61b2c08fcaf7b 100644
--- a/dist/hls.js
+++ b/dist/hls.js	
@@ -17943,6 +17943,7 @@
         hasAudio: false,
         hasVideo: true,
         nb: outputSamples.length,
+        outputSamples: outputSamples,
         dropped: track.dropped
       };
       // For troubleshooting duplicates of https://github.com/video-dev/hls.js/issues/6374
@@ -18179,7 +18180,8 @@
         type: type,
         hasAudio: true,
         hasVideo: false,
-        nb: nbSamples
+        nb: nbSamples,
+        outputSamples: outputSamples
       };
       this.isAudioContiguous = true;
       return audioData;
diff --git a/dist/hls.mjs b/dist/hls.mjs
index fce9ba311355708eaf3846cc9f167a1fdd946c76..98b4e3888343303377d7ae27ed28bcfffb9e0013 100644
--- a/dist/hls.mjs
+++ b/dist/hls.mjs	
@@ -11182,6 +11182,10 @@ class BaseStreamController extends TaskLoop {
         // Buffer must be ahead of first part + duration of parts after last segment
         // and playback must be at or past segment adjacent to part list
         const firstPart = details.partList[0];
+        // Loading of VTT subtitle parts is not implemented in subtitle-stream-controller (#7460)
+        if (firstPart.fragment.type === PlaylistLevelType.SUBTITLE) {
+          return false;
+        }
         const safePartStart = firstPart.end + (((_details$fragmentHint = details.fragmentHint) == null ? void 0 : _details$fragmentHint.duration) || 0);
         if (bufferEnd >= safePartStart) {
           const playhead = this.playhead;
@@ -17371,6 +17375,7 @@ class MP4Remuxer extends Logger {
       hasAudio: false,
       hasVideo: true,
       nb: outputSamples.length,
+      outputSamples,
       dropped: track.dropped
     };
     // For troubleshooting duplicates of https://github.com/video-dev/hls.js/issues/6374
@@ -17605,7 +17610,8 @@ class MP4Remuxer extends Logger {
       type,
       hasAudio: true,
       hasVideo: false,
-      nb: nbSamples
+      nb: nbSamples,
+      outputSamples
     };
     this.isAudioContiguous = true;
     return audioData;
@@ -41139,5 +41145,5 @@ class Hls {
 }
 Hls.defaultConfig = void 0;
 
-export { AbrController, AttrList, AudioStreamController, AudioTrackController, BaseLoader, BasePlaylistController, BaseSegment, BaseStreamController, BufferController, CMCDController, CapLevelController, ChunkMetadata, ContentSteeringController, Cues, DateRange, EMEController, ErrorActionFlags, ErrorController, ErrorDetails, ErrorTypes, Events, FPSController, FetchLoader, Fragment, Hls, HlsSkip, HlsUrlParameters, KeySystemFormats, KeySystems, Level, LevelDetails, LevelKey, LoadStats, LoaderContextType, M3U8Parser, MetadataSchema, NetworkErrorAction, Part, PlaylistLevelType, SubtitleStreamController, SubtitleTrackController, TimelineController, XhrLoader, Hls as default, fetchSupported, getMediaSource, isMSESupported, isSupported, requestMediaKeySystemAccess };
+export { AACDemuxer, AbrController, AttrList, AudioStreamController, AudioTrackController, BaseLoader, BasePlaylistController, BaseSegment, BaseStreamController, BufferController, CMCDController, CapLevelController, ChunkMetadata, ContentSteeringController, Cues, DateRange, EMEController, ErrorActionFlags, ErrorController, ErrorDetails, ErrorTypes, Events, FPSController, FetchLoader, Fragment, Hls, HlsSkip, HlsUrlParameters, KeySystemFormats, KeySystems, Level, LevelDetails, LevelKey, LoadStats, LoaderContextType, M3U8Parser, MP3Demuxer, MP4Demuxer, MP4Remuxer, MetadataSchema, NetworkErrorAction, Part, PassThroughRemuxer, PlaylistLevelType, SubtitleStreamController, SubtitleTrackController, TSDemuxer, TimelineController, XhrLoader, Hls as default, fetchSupported, getMediaSource, isMSESupported, isSupported, requestMediaKeySystemAccess };
 //# sourceMappingURL=hls.mjs.map
